### ISSUE TITLE
Some ideas regarding DHCP overlay

### ISSUE THREAD

**Issue by brokoler** [TechnitiumSoftware/DnsServer#837]
Hello,

still loving Technitium and recommending it to everyone!

Just had some ideas how the DHCP overlay could be even better:
1. Editable hostnames in DHCP Leases table
2. Add multiple Leases Tabs (One per DHCP scope)

This would both allow to get a better overview over DHCP clients.
In networks with multiple DHCP scopes and multiple subnets it's sometimes hard to follow.
Also many clients don't announce their hostname via network or it's not in an understandable format.

For 1. I think there is place for this, since only two tabs are used so far?
In this case the "Scopes" tab could be moved to the left, since it's the configuration option.
And from there each Scope could be listed separately on the right side, for the name of the Leases Table the DHCP scope name could be used.

![grafik](https://github.com/TechnitiumSoftware/DnsServer/assets/57031979/8386913b-6f1f-4e8f-92ce-e7d1aa6bc405)

For 2. the announced DHCP hostname could be cached?
Maybe it would be possible to have an "Edit" button besides each hostname, entering a custom hostname would show the new name everywhere? If the hostname field is empty, it could fall back to the default hostname announced by the device and if the name was customized a small sign besides the hostname (Like an asterisk *) that shows it was edited and is not the default value?

Really interested what you think about these suggestions @ShreyasZare 
Thanks so much for keeping this software great!

**Comment by ShreyasZare** [comment]
Thanks for the suggestions and contribution!

Filtering option is on my to-do list already for DHCP leases and even for Zones section and zone records section. It will take a while as I add features but surely will be implemented.

Edit option for hostnames in lease looks interesting. The only issue I can think of now is that the updated hostname wont be reflected until the host renews lease, so the changed hostname that will be visible in the list wont be of much use. May be the DNS entries can be updated but the host itself will remain unaware of this until the renewal. Will think on this when DHCP major upgrade is planned. Adding options to leases list will be really useful since its easy to access and will make updating multiple entries fast so the lease view is good place to have more options specially combined with filtering.
